File: packages/swift/Sources/CatalystSwift/Models/CatalystCreateAlbumBookRequest.swift
// Licensed under the MIT License
//
// Copyright (c) 2025 Natsune Mochizuki

/// Request body shared by `POST /catalyst/v1/album/by/id/{id}/book` and
/// `POST /catalyst/v1/smart-album/by/id/{id}/book`.
///
/// `template` is one of "photo-book" | "record-book".
/// `quality` is one of "web" | "print".
/// `tocType` is one of "per-post" | "page-only" | nil.
public struct CatalystCreateAlbumBookRequest: Encodable, Sendable {
  public let template: String
  public let quality: String
  public let coverImageUrl: String?
  public let subtitle: String?
  public let customText: String?
  public let showBody: Bool?
  public let showDate: Bool?
  public let showUrl: Bool?
  public let showQrCode: Bool?
  public let tocType: String?
  public let colophonText: String?

  public init(
    template: String,
    quality: String,
    coverImageUrl: String? = nil,
    subtitle: String? = nil,
    customText: String? = nil,
    showBody: Bool? = nil,
    showDate: Bool? = nil,
    showUrl: Bool? = nil,
    showQrCode: Bool? = nil,
    tocType: String? = nil,
    colophonText: String? = nil
  ) {
    self.template = template
    self.quality = quality
    self.coverImageUrl = coverImageUrl
    self.subtitle = subtitle
    self.customText = customText
    self.showBody = showBody
    self.showDate = showDate
    self.showUrl = showUrl
    self.showQrCode = showQrCode
    self.tocType = tocType
    self.colophonText = colophonText
  }
}

File: packages/dotnet/CatalystSharp/Models/Requests/CatalystCreateAlbumBookRequest.cs
using System.Text.Json.Serialization;

namespace CatalystSharp.Models.Requests;

// template: "photo-book" | "record-book"
// quality: "web" | "print"
// tocType: "per-post" | "page-only" | null
public record CatalystCreateAlbumBookRequest(
    [property: JsonPropertyName("template")] string Template,
    [property: JsonPropertyName("quality")] string Quality,
    [property: JsonPropertyName("coverImageUrl")] string? CoverImageUrl = null,
    [property: JsonPropertyName("subtitle")] string? Subtitle = null,
    [property: JsonPropertyName("customText")] string? CustomText = null,
    [property: JsonPropertyName("showBody")] bool? ShowBody = null,
    [property: JsonPropertyName("showDate")] bool? ShowDate = null,
    [property: JsonPropertyName("showUrl")] bool? ShowUrl = null,
    [property: JsonPropertyName("showQrCode")] bool? ShowQrCode = null,
    [property: JsonPropertyName("tocType")] string? TocType = null,
    [property: JsonPropertyName("colophonText")] string? ColophonText = null
);
